src/platforms/windows/daemon/windowssplittunnel.cpp

```diff
@@ -144,96 +150,112 @@ ProcessInfo getProcessInfo(HANDLE process, const PROCESSENTRY32W& processMeta) {
 
 }  // namespace
 
-WindowsSplitTunnel::WindowsSplitTunnel(QObject* parent) : QObject(parent) {
+std::unique_ptr<WindowsSplitTunnel> WindowsSplitTunnel::create(
+    WindowsFirewall* fw) {
+  // 00: Check if we conflict with mullvad, if so.
   if (detectConflict()) {
     logger.error() << "Conflict detected, abort Split-Tunnel init.";
-    uninstallDriver();
-    return;
+    return nullptr;
   }
+  // 01: Check if the driver is installed, if not do so.
   if (!isInstalled()) {
     logger.debug() << "Driver is not Installed, doing so";
     auto handle = installDriver();
     if (handle == INVALID_HANDLE_VALUE) {
       WindowsUtils::windowsLog("Failed to install Driver");
-      return;
+      return nullptr;
     }
     logger.debug() << "Driver installed";
     CloseServiceHandle(handle);
   } else {
-    logger.debug() << "Driver is installed";
-  }
-  initDriver();
-}
-
-WindowsSplitTunnel::~WindowsSplitTunnel() {
-  CloseHandle(m_driver);
-  uninstallDriver();
-}
-
-void WindowsSplitTunnel::initDriver() {
-  if (detectConflict()) {
-    logger.error() << "Conflict detected, abort Split-Tunnel init.";
-    return;
-  }
-  logger.debug() << "Try to open Split Tunnel Driver";
-  // Open the Driver Symlink
-  m_driver = CreateFileW(DRIVER_SYMLINK, GENERIC_READ | GENERIC_WRITE, 0,
-                         nullptr, OPEN_EXISTING, 0, nullptr);
+    logger.debug() << "Driver was installed";
+  }
+  // 02: Now check if the service is running
+  auto driver_manager =
+      WindowsServiceManager::open(QString::fromWCharArray(DRIVER_SERVICE_NAME));
+  if (Q_UNLIKELY(driver_manager == nullptr)) {
+    // Let's be fair if we end up here,
+    // after checking it exists and installing it,
+    // this is super unlikeley
+    Q_ASSERT(false);
+    logger.error()
+        << "WindowsServiceManager was unable fo find Split Tunnel service?";
+    return nullptr;
+  }
+  if (!driver_manager->isRunning()) {
+    logger.debug() << "Driver is not running, starting it";
+    // Start the service
+    if (!driver_manager->startService()) {
+      logger.error() << "Failed to start Split Tunnel Service";
+      return nullptr;
+    };
+  }
+  // 03: Open the Driver Symlink
+  auto driverFile = CreateFileW(DRIVER_SYMLINK, GENERIC_READ | GENERIC_WRITE, 0,
+                                nullptr, OPEN_EXISTING, 0, nullptr);
   ;
-
-  if (m_driver == INVALID_HANDLE_VALUE) {
+  if (driverFile == INVALID_HANDLE_VALUE) {
     WindowsUtils::windowsLog("Failed to open Driver: ");
+    return nullptr;
+  }
+  // 04: Make sure the Windows Firewall has created the sublayer
+  // otherwise the driver will fail to initialize
+  if (!fw || !fw->init()) {
```

Comment:
Minor nit: I get that this hasn't really changed, but it is kindof odd for the firewall init to be happening here. It would be a little more appropriate for `WindowsDaemon` to do the init, and then pass in a reference to the WFP sublayer as an argument to `WindowsSplitTunnel::create()`

Reply:
How exactly does the split tunnel driver figure out the WFP sublayer to use?

Reply:
The Sublayer is hardcoded in the driver...